swan-lake/learn/testing-ballerina-code/writing-tests.md

```diff
@@ -355,6 +394,82 @@ function testFunction3() {
     test:assertTrue(true, msg = "Failed!");
 }
 ```
+The function specified following the annotation will be run after each test within the test suite is run. This can be
+ used for repeatedly cleaning up test-level aspects before every test function.
+
+### @test:AfterEach
+
+***Example:***
+
+```ballerina
+import ballerina/io;
+import ballerina/test;
+
+// After each function, which is executed before each test function
+@test:AfterEach
+function beforeFunc() {
+    io:println("I'm the before function!");
+}
+
+// Test function
+@test:Config {}
+function testFunction1() {
+    io:println("I'm in test function 1!");
+    test:assertTrue(true, msg = "Failed!");
+}
+
+// Test function
+@test:Config {}
+function testFunction2() {
+    io:println("I'm in test function 2!");
+    test:assertTrue(true, msg = "Failed!");
+}
+
+// Test function
+@test:Config {}
+function testFunction3() {
+    io:println("I'm in test function 3!");
+    test:assertTrue(true, msg = "Failed!");
+}
+```
+
+## @test:AfterGroups {}
+
+***Example:***
+
+The function specified following the annotation will be run will execute after the first test belonging to the
```

Comment:
```suggestion
Once the function specified after the annotation will be run, it will be executed after the first test belonging to the
```
Meaning is not clear.

Reply:
Addressed in af5c8a9